lib/vfs/src/builder.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{FileSystem, VirtualFile};
+use std::path::{Path, PathBuf};
+use tracing::*;
+use wasmer_wasi_types::types::{__WASI_STDERR_FILENO, __WASI_STDIN_FILENO, __WASI_STDOUT_FILENO};
+
+use super::ZeroFile;
+use super::{NullFile, SpecialFile};
+use crate::tmp_fs::TmpFileSystem;
+
+pub struct RootFileSystemBuilder {
+    default_root_dirs: bool,
+    default_dev_files: bool,
+    add_wasmer_command: bool,
+    stdin: Option<Box<dyn VirtualFile + Send + Sync>>,
+    stdout: Option<Box<dyn VirtualFile + Send + Sync>>,
+    stderr: Option<Box<dyn VirtualFile + Send + Sync>>,
+    tty: Option<Box<dyn VirtualFile + Send + Sync>>,
+}
```

Comment:
Why is there a tty & the 3 std stream?

Reply:
Because tty files in /dev/tty act different to /dev/stdout. Stdout inherits stdio from the parent while tty bypasses the process tree and connects stdio directly to the virtual operating system

Reply:
Assumed to be answered - it's required. (@ptitSeb )